h2o-r/h2o-package/R/xgboost.R

```diff
@@ -599,3 +599,46 @@ h2o.xgboost.available <- function() {
     }
 }
 
+#' XGBoost feature interactions and importance, leaf statistics and split value histograms in a tabular form.
+#'
+#' Metrics:
+#' Gain - Total gain of each feature or feature interaction.
+#' FScore - Amount of possible splits taken on a feature or feature interaction.
+#' wFScore - Amount of possible splits taken on a feature or feature interaction weighed by 
+#' the probability of the splits to take place.
+#' Average wFScore - wFScore divided by FScore.
+#' Average Gain - Gain divided by FScore.
+#' Expected Gain - Total gain of each feature or feature interaction weighed by the probability to gather the gain.
+#' Average Tree Index
+#' Average Tree Depth
+#'
+#' @param max_interaction_depth Upper bound for extracted feature interactions depth. Defaults to 100.
+#' @param max_tree_depth: Upper bound for tree depth. Defaults to 100.
+#' @param max_deepening: Upper bound for interaction start deepening (zero deepening => interactions 
+#' starting at root only). Defaults to -1.
+#'
+#' @examples
+#' \dontrun{
+#' library(h2o)
+#' h2o.init()
+#' boston <- h2o.importFile(
+#'        "https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/BostonHousing.csv",
+#'         destination_frame="boston"
+#'         )
+#' boston_xgb <- h2o.xgboost(training_frame = boston, y = "medv", seed = 1234)
+#' feature_interactions <- h2o.xgboost.feature_interaction(boston_xgb)
+#' }
+#' @export
+h2o.xgboost.feature_interaction <- function(model, max_interaction_depth = 100, max_tree_depth = 100, max_deepening = -1) {
```

Comment:
are other models going to be supported in the future? if yes, than we should create a more generic api

Reply:
xgboost and gbm

Reply:
Since the API and back-end will be the same for both xgboost and gbm I think we should have just one function for both. But I am not sure. Maybe @ledell can provide some feedback?
Also I think the norm now is to use camelCase for naming R APIs.

Reply:
the requirement was to do this for XGB as it is done in XGBFI package (this part is clear), but XGBFI isn't intended for GBM and I am not sure whether it can be adjusted to use it. For gbm feature importance there is a paper on how to measure it by calculating H-statistic what is a different approach, so if used, might be that the gbm function will have different signature. So I am not sure whether to prepare more generic API in case it will be possible to adjust XGBFI approach to gbm - when it may happen that I will not use it in the end. In case I will, I can always add more generic and deprecate and after some time remove this specific xgb one. I am not sure what approach will be used for gbm right now.